pkg/arvo/sur/graph-store.hoon

```diff
@@ -0,0 +1,69 @@
+/-  *post
+|%
+++  mop
+  |*  [key=mold value=mold]
+  |=  ord=$-([key key] ?)
+  |=  a=*
+  =/  b  ;;((tree [key=key val=value]) a)
+  ?>  (check-balance:((ordered-map key value) ord) b)
+  b
+::
++$  mark          term
++$  graph         ((mop atom node) gth)
++$  marked-graph  [p=graph q=(unit mark)]
+::
++$  node          [=post children=internal-graph]
++$  graphs        (map resource marked-graph)
+::
++$  tag-queries   (jug term resource)
+::
++$  update-log    ((mop time logged-update) gth)
++$  update-logs   (map resource update-log)
+::
++$  internal-graph
```

Comment:
This should just be `(unit graph)`.

Reply:
It doesn't compile with `+$  internal-graph  (unit graph)`, hence why I wrote it this way. I tried again once you commented and it still doesn't compile.
<img width="753" alt="Screen Shot 2020-07-20 at 1 58 06 PM" src="https://user-images.githubusercontent.com/1377557/87985814-16371a00-ca91-11ea-93c6-db0216866b60.png">
<img width="752" alt="Screen Shot 2020-07-20 at 1 58 16 PM" src="https://user-images.githubusercontent.com/1377557/87985824-1afbce00-ca91-11ea-8a3e-560191f8273b.png">

Reply:
Have you tried putting them in separate cores? Tends to resolve shenanigans like this.

Reply:
I'm trying to do that currently, but I can't imagine a way to do it that would allow this dependency cycle to resolve, since both need to be in the subject for either to compile.
```
    +$  graph         ((mop atom node) gth)
    +$  node          [=post children=(unit graph)]
```

Reply:
It doesn't surprise me that unit gets confused with this.  I think the way you're doing it is fine.